src/Comparator/Variance/TypeIsContravariant.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\ApiCompare\Comparator\Variance;
+
+use Roave\BetterReflection\Reflection\ReflectionType;
+use function in_array;
+use function strtolower;
+
+/**
+ * This is a simplistic contravariant type check. A more appropriate approach would be to
+ * have a `$type->includes($otherType)` check with actual types represented as value objects,
+ * but that is a massive piece of work that should be done by importing an external library
+ * instead, if this class no longer suffices.
```

Comment:
Issue for this?

Reply:
Not really: I think we're fine with this for now, but we will hit edge cases when we start supporting union, intersection and nested types (`(Foo[]|Bar)[][]` for example) in #44 and #45. Luckily, property types are invariant, so any type change is a BC break :D